fvm/src/kernel/mod.rs

```diff
@@ -80,15 +59,42 @@ pub trait Kernel:
         actor_id: ActorID,
         method: MethodNum,
         value_received: TokenAmount,
-        execution_type: ExecutionType,
     ) -> Self
     where
-        Self: Sized;    
+        Self: Sized;
 }
 
-/// TODO
-pub trait Validator {
-    fn is_validator(&self) -> bool;
+/// The "kernel" implements the FVM interface as presented to the actors. It:
+///
+/// - Manages the Actor's state.
+/// - Tracks and charges for IPLD & syscall-specific gas.
+///
+/// Actors may call into the kernel via the syscalls defined in the [`syscalls`][crate::syscalls]
+/// module.
+pub trait Kernel:
+    BaseKernel
+    + ActorOps
+    + IpldBlockOps
+    + CircSupplyOps
+    + GasOps
+    + MessageOps
+    + NetworkOps
+    + RandomnessOps
+    + SelfOps
+    + SendOps
+{
+}
+
+pub trait CheckedKernel: BaseKernel {
```

Comment:
`ValidationKernel`?

Reply:
`ValidationKernel` is what I went with originally, though this trait will also be used for abstract accounts under normal execution. So this needs some further iteration anyways.

Reply:
Hm. _That_ seems like it isn't going to work because it won't compose. I.e., a single actor may have one or more "properties" that the kernel may need to react to.

Basically, this comes down to "modes" versus "attributes":

- We may be invoking an actor in a specific "mode". In that case, this mode can be reflected in the kernel.
- We may be invoking an actor with one or more attributes (e.g., is abstract account, is system actor, etc.). These likely aren't going to be exclusive.